inbox/events/microsoft/parse.py

```diff
@@ -81,6 +81,16 @@ def parse_msgraph_datetime_tz_as_utc(datetime_tz: MsGraphDateTimeTimeZone):
     # so we need to use ciso8601 because Python stdlib only supports up to 6.
     dt = ciso8601.parse_datetime(datetime_tz["dateTime"])
 
+    # Microsoft sometimes returns payloads such as this one:
+    # {"dateTime": "0001-01-01T00:00:00.0000000Z", "timeZone": "tzone://Microsoft/Utc"}
+    # Notice the `Z` - the `dateTime` is timezone-aware. This is undocumented
+    # behaviour, but we will handle it here so it doesn't break sync.
+    if dt.tzinfo:
+        # We will only humour Microsoft when it comes to UTC. This can be
```

Comment:
```suggestion
        # We will only honour Microsoft when it comes to UTC. This can be
```

Reply:
I wrote `humour` on purpose 😅 We're humouring Microsoft in that we're letting them return inconsistent data, and we're trying to process it without failing.